ext/storage_sources/s3_store/test/test_s3_connection.cpp

```diff
@@ -250,20 +299,21 @@ TestObjectExists(const Aws::S3Crt::ClientConfiguration &config)
     File << "Test payload";
     File.close();
 
-    if (ret = conn.ObjectExists(objectName, exists) != 0 || exists)
+    ret = conn.ObjectExists(objectName, exists);
+    if (ret != 0 || exists)
         return (ret);
```

Comment:
If the test fails here because the file unexpectedly exists, the return will be 0 from the ObjectExists() call. A 0 return will imply a success.